pkg/eventing/v1alpha1/client_test.go

```diff
@@ -0,0 +1,116 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"fmt"
+	"testing"
+
+	"gotest.tools/assert"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	client_testing "k8s.io/client-go/testing"
+	"knative.dev/eventing/pkg/apis/eventing/v1alpha1"
+	"knative.dev/eventing/pkg/client/clientset/versioned/typed/eventing/v1alpha1/fake"
+)
+
+var testNamespace = "test-ns"
+
+func setup() (sources fake.FakeEventingV1alpha1, client KnEventingClient) {
+	sources = fake.FakeEventingV1alpha1{Fake: &client_testing.Fake{}}
+	client = NewKnEventingClient(&sources, testNamespace)
+	return
+}
+
+func TestDeleteTrigger(t *testing.T) {
```

Comment:
I have negative path test in this unit test. Yet since it's a kind of fake test, it's not helpful to add twice deletes with error. I will add twice deletes in the integration test case.

Reply:
I added a delete with error test case for no existing trigger using mock client.